ui/components/institutional/qr-code-modal/qr-code-modal.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect, useContext } from 'react';
+import PropTypes from 'prop-types';
+import QRCode from 'qrcode.react';
+import { Modal, ModalOverlay, Text, Box } from '../../component-library';
+import { ModalContent } from '../../component-library/modal-content/modal-content';
+import { ModalHeader } from '../../component-library/modal-header/modal-header';
+import {
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { I18nContext } from '../../../contexts/i18n';
+import Spinner from '../../ui/spinner';
+
+export default function QRCodeModal({ onClose, custodianName }) {
+  const t = useContext(I18nContext);
+  const [publicKeyData, setPublicKeyData] = useState(null);
+  const [error, setError] = useState('');
+
+  async function generatePublicKey() {
+    try {
+      const { publicKey } = await window.crypto.subtle.generateKey(
+        {
+          name: 'RSA-OAEP',
+          modulusLength: 2048,
+          publicExponent: new Uint8Array([0x01, 0x00, 0x01]),
+          hash: 'SHA-256',
+        },
+        true,
+        ['encrypt', 'decrypt'],
+      );
+
+      const exportedPublicKey = await window.crypto.subtle.exportKey(
+        'spki',
+        publicKey,
+      );
+
+      const publicKeyBase64 = Buffer.from(exportedPublicKey)
+        .toString('base64')
+        .replace(/\+/gu, '-')
```

Comment:
@albertolive what does this replace and the others do? just wondering 😅

Reply:
Yes, after revisiting it doesn't make sense to clean up the exportedPublikKey. I will send it as it is because then the backend doesn't need to undo what I did. The idea behind this was to send a shorter string in the QR code